### PR TITLE
Miscellaneous Travis improvements

### PR DESCRIPTION
**Pull request by vadz**

### CODE REVIEW

**Comment by tierra** [comment]
Nice. I've been wanting to get OSX in the mix here for a while. Same with some trusty builds. Hopefully you have some good luck with this.

**Comment by vadz** [comment]
It mostly seems to work fine, but I just can't convince homebrew to install the universal (or even just 64 bit) version of cppunit somehow... I must be missing something because, if anything, it should install the 64 bit version _only_, but I see no other explanation: there is `-L/usr/local/Cellar/cppunit/1.13.2/lib -lcppunit` in the linker command line yet all cppunit symbols are not found as can be seen [in the log](https://travis-ci.org/wxWidgets/wxWidgets/jobs/99520780)

File: build/tools/before_install.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    Linux)
+        if [ -f /etc/apt/sources.list ]; then
+            $SUDO apt-get update
+            $SUDO apt-get install -y libcppunit-dev libgtk2.0-dev libnotify-dev
+        fi
+        ;;
+
+    Darwin)
+        brew update
+        brew install --universal cppunit
```
Comment (tierra) [inline build/tools/before_install.sh]: I think this flag is required to come *after* the package, so:
```brew install cppunit --universal```

**Comment by vadz** [comment]
Unfortunately putting `--universal` at the end didn't help neither.

**Comment by vadz** [comment]
I suspect the problem with cppunit was actually due to using the wrong C++ library due to the wrong use of 10.5 as minimal OS X version and this was fixed in c3504663efad901677155e484f2b00fa0fce7fc8, so hopefully everything should work now. Let's see...

**Pull request merged**